frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageInstructionsTimer.vue

```diff
@@ -0,0 +1,191 @@
+      <v-btn
+        icon
+        :disabled="timer.remainingTime.value <= 30"
+        depressed
+        @click="timer.remainingTime.value -= 30"><v-icon>mdi-minus</v-icon></v-btn>
+      {{ timer.hours.value ? timer.hours.value + "h" : "" }} {{ timer.minutes.value ? timer.minutes.value + "m" : "" }}
+      {{ (timer.seconds.value || !timer.minutes.value) ? timer.seconds.value + "s" : "" }}
+      <v-btn
+        icon
+        depressed
+        @click="timer.remainingTime.value += 30"><v-icon>mdi-plus</v-icon></v-btn>
+      <v-btn
+        v-if="!timer.isCounting.value && !timer.hasStarted.value"
+        rounded
+        depressed
+        @click="timer.startAlarm"> {{$t("recipe.timer.start")}} </v-btn>
+      <v-btn
+        v-else-if="(timer.hasStarted.value && timer.isCounting.value)"
+        rounded
+        depressed
+        @click="timer.pauseTimer"> {{$t("recipe.timer.pause")}} </v-btn>
+      </v-btn>
+      <span v-else-if="!timer.done.value">
+        <v-btn
+          rounded
+          depressed
+          @click="timer.startAlarm">
+          {{$t("recipe.timer.continue")}}
+        </v-btn>
+        <v-btn
+          icon
+          @click="timer.resetTimer"><v-icon>mdi-restore</v-icon></v-btn>
+      </span>
+      <span v-else>
+        <v-btn
+          icon
+          @click="timer.resetTimer"><v-icon>mdi-restore</v-icon></v-btn>
+      </span>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+  import { computed, ref, onBeforeUnmount } from "vue";
+import { useContext } from "@nuxtjs/composition-api";
+
+  import { TranslateResult } from "vue-i18n"
+  // @ts-ignore typescript can't find our audio file, but it's there!
+  import timerAlarmAudio from "~/assets/audio/kitchen_alarm.mp3";
+  export default {
+    props: {
+      text: {
+        type: String,
+        default: "",
+      },
+    },
+    setup(props) {
+      const { i18n } = useContext();
+      // Makes regex work with multiple languages via locales
+      const minuetStrings = computed<string[]>(() => {
+        console.log(i18n.availableLocales);
+        const result: TranslateResult[] = [];
+        i18n.availableLocales.forEach((locale) => {
+          result.push(i18n.t("recipe.timer.minute", locale ))
+          result.push(i18n.t("recipe.timer.minutes", locale ))
+        })
+        return result
+      })
+
+      const extractMinutes = (text: string) => {
+        const regexMins = new RegExp(`\\b(\\d+)\\s*(${minuetStrings.value.join("|")})\\b`, "gi");
+        const matchesMin = text.matchAll(regexMins);
+        const timers = [];
+        for (const match of matchesMin) {
+          timers.push(createTimer(Number(match[1]) * 60));
+        }
+        return timers;
+      };
+      const timers = computed(() => {
+        return extractMinutes(props.text as string);
+      });
+
+      const createTimer = (totalSeconds: number) => {
```

Comment:
A lot of this code is re-used from `RecipeTimerMenu`. Can we extract this logic into a composable so we don't have duplicate code?

Reply:
Probably, are you thinking something that just generates the functions and state values and returns them as a object?

Reply:
Yeah basically keep all of the state management in the composable and only expose a bare-minimum API for the components to use. Something like:

```json
export const useTimer = function () => {
  ...  // your time code
  const controls = {
    updateTimer,
    pauseTimer,
    resetTimer,
    ...
  }
  return { controls }  // or anything else the consumer might want
}
```

Additionally, you could potentially combine `RecipePageInstructionsTimer` and `RecipeTimerMenu` into one component (or two components with a shared sub component). But either way we should probably move the logic/state into a composable.

Reply:
You can take a look at our store factory to see kind of what I'm getting at: https://github.com/mealie-recipes/mealie/blob/mealie-next/frontend/composables/partials/use-store-factory.ts

Reply:
I have extracted the timer logic from the existing timer, adjusted it a little and reused it in both timer.